profiling/src/internal/profile/fuzz_tests.rs

```diff
@@ -0,0 +1,631 @@
+        Self {
+            locations: value.locations.iter().map(api::Location::from).collect(),
+            values: value.values.clone(),
+            labels: value.labels.iter().map(api::Label::from).collect(),
+        }
+    }
+}
+
+/// PartialEq and Hash work on just the key to avoid getting duplicate
+/// label keys.
+impl PartialEq for Label {
+    fn eq(&self, other: &Self) -> bool {
+        self.key == other.key
+    }
+}
+
+impl Eq for Label {}
+
+/// PartialEq and Hash work on just the key to avoid getting duplicate
+/// label keys.
+impl std::hash::Hash for Label {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.key.hash(state)
+    }
+}
+impl TypeGenerator for Sample {
+    fn generate<D: bolero_generator::Driver>(driver: &mut D) -> Option<Self> {
+        let locations = Vec::<Location>::gen_with().generate(driver)?;
+        let values = Vec::<i64>::gen_with().generate(driver)?;
+        let mut labels = std::collections::HashSet::<Label>::gen_with().generate(driver)?;
+
+        // Ensure that the label has "local root span id" key
+        // Generate non-zero num value for the label
+        let num = i64::gen().generate(driver)?;
+        // zero num is considered as invalid, see Profile::validate_sample_labels
+        if num == 0 {
+            return None;
+        }
+
+        labels.insert(Label {
+            key: "local root span id".into(),
+            str: None,
+            num,
+            num_unit: None,
+        });
+        let labels = labels.into_iter().collect();
+        Some(Self {
+            locations,
+            values,
+            labels,
+        })
+    }
+}
+
+fn assert_sample_types_eq(
+    profile: &pprof::Profile,
+    expected_sample_types: &[owned_types::ValueType],
+) {
+    assert_eq!(
+        profile.sample_types.len(),
+        expected_sample_types.len(),
+        "Sample types length mismatch"
+    );
+    for (typ, expected_typ) in profile
+        .sample_types
+        .iter()
+        .zip(expected_sample_types.iter())
+    {
+        assert_eq!(*profile.string_table_fetch(typ.r#type), *expected_typ.r#typ);
+        assert_eq!(*profile.string_table_fetch(typ.unit), *expected_typ.unit);
+    }
+}
+
+fn assert_samples_eq(
+    profile: &pprof::Profile,
+    samples_with_timestamps: &[&Sample],
+    samples_without_timestamps: &HashMap<(&Vec<Location>, &Vec<Label>), Vec<i64>>,
+    endpoint_mappings: &FxIndexMap<u64, &String>,
+) {
+    assert_eq!(
+        profile.samples.len(),
+        samples_with_timestamps.len() + samples_without_timestamps.len(),
+        "Samples length mismatch"
+    );
+
+    let mut expected_timestamped_samples = samples_with_timestamps.iter();
+
+    for sample in profile.samples.iter() {
+        // Recreate owned_locations from vector of pprof::Location
+        let mut owned_locations = Vec::new();
+        for loc_id in sample.location_ids.iter() {
+            let location = &profile.locations[*loc_id as usize - 1];
```

Comment:
comment explaining the `-1`

Reply:
`[internal::Location]` is created via its `from_offset()` function and the function uses `small_non_zero_pprof_id()`.

https://github.com/DataDog/libdatadog/blob/f3d0382d1c7c43dde7e22d3b27aad270d146bcee/profiling/src/internal/location.rs#L48

The function's documentation says 
```
/// Creates a non-zero, 32-bit unsigned id from the offset. It's guaranteed to
/// be the offset + 1, with guards to not overflow the size of u32.
///
/// This is useful because many pprof collections do not allow an item with an
/// id of zero, even if it's the first item in the collection.
```